core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -126,28 +125,11 @@ public BlockWorkerClient acquireWorkerClient() {
     BlockWorkerClient client = acquireLocalWorkerClient();
     if (client == null) {
       // Get a worker client for any worker in the system.
-      return acquireRemoteWorkerClient("");
-    }
-    return client;
-  }
-
-  /**
-   * Obtains a worker client to the worker with the given hostname in the system.
-   *
-   * @param hostname the hostname of the worker to get a client to, empty String indicates all
-   *        workers are eligible
-   * @return a {@link BlockWorkerClient} connected to the worker with the given hostname
-   * @throws IOException if no Alluxio worker is available for the given hostname
-   */
-  public BlockWorkerClient acquireWorkerClient(String hostname) throws IOException {
-    BlockWorkerClient client;
-    if (hostname.equals(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()))) {
-      client = acquireLocalWorkerClient();
-      if (client == null) {
-        throw new IOException(ExceptionMessage.NO_WORKER_AVAILABLE_ON_HOST.getMessage(hostname));
+      List<WorkerNetAddress> workerAddresses = getWorkerAddresses("");
+      if (workerAddresses.isEmpty()) {
+        return acquireRemoteWorkerClient(null);
       }
-    } else {
-      client = acquireRemoteWorkerClient(hostname);
+      return acquireRemoteWorkerClient(workerAddresses.get(0));
```

Comment:
In fact, this method is not used anymore, because the client uses location policy to acquire the worker.